core/schema/modulesource.go

```diff
@@ -213,16 +214,25 @@ type buildkitClient interface {
 // - if not, try to isolate root of git repo from the ref
 // - if nothing worked, fallback as local ref, as before
 func parseRefString(ctx context.Context, bk buildkitClient, refString string) parsedRefString {
+	ctx, span := core.Tracer(ctx).Start(ctx, fmt.Sprintf("parseRefString: %s", refString))
+
+	defer span.End()
 	localParsed := parsedRefString{
 		modPath: refString,
 		kind:    core.ModuleSourceKindLocal,
 		scheme:  core.NoScheme,
 	}
 
+	if strings.HasPrefix(refString, ".") {
+		return localParsed
+	}
```

Comment:
suggestion: *might* be worth a comment on explaining why this optimization works.